src/libraries/System.Private.CoreLib/src/System/OperatingSystem.cs

```diff
@@ -172,18 +177,22 @@ public static bool IsAndroidVersionAtLeast(int major, int minor = 0, int build =
             => IsAndroid() && IsOSVersionAtLeast(major, minor, build, revision);
 
         /// <summary>
-        /// Indicates whether the current application is running on iOS.
+        /// Indicates whether the current application is running on iOS or MacCatalyst.
         /// </summary>
+        [SupportedOSPlatformGuard("ios")]
```

Comment:
I would prefer to remove it as we are discussed to not make the analyzer fully dependent on Guard attributes (at least not within 6.0)
```suggestion
```

Reply:
Oh, and literally only apply the guard attributes for _additional_ platforms? Makes sense.